pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -1266,13 +1259,31 @@ func getDefaultQueue(append bool, defaultJmsQueue string, jmsQueue string) strin
 	return ""
 }
 
-func setPasswords(spec *api.KieAppSpec, isTrialEnv bool) {
+func setPasswords(spec *api.KieAppSpec, status api.KieAppStatus, isTrialEnv bool) {
 	passwords := []*string{
 		&spec.CommonConfig.KeyStorePassword,
 		&spec.CommonConfig.DBPassword,
 		&spec.CommonConfig.AMQPassword,
 		&spec.CommonConfig.AMQClusterPassword,
 	}
+
+	if len(spec.CommonConfig.SecretAdminCredentials) == 0 {
+		if len(status.Applied.CommonConfig.AdminPassword) == 0 {
+			passwords = append(passwords, &spec.CommonConfig.AdminPassword)
+		} else if len(spec.CommonConfig.AdminPassword) == 0 {
+			spec.CommonConfig.AdminPassword = status.Applied.CommonConfig.AdminPassword
+		}
+	}
+
+	if spec.Objects.Console != nil && spec.Objects.Console.DataGridAuth != nil && len(spec.Objects.Console.DataGridAuth.Password) == 0 {
+		if status.Applied.Objects.Console != nil && status.Applied.Objects.Console.DataGridAuth != nil && len(status.Applied.Objects.Console.DataGridAuth.Password) > 0 {
+			if len(spec.Objects.Console.DataGridAuth.Password) == 0 {
+				spec.Objects.Console.DataGridAuth.Password = status.Applied.Objects.Console.DataGridAuth.Password
+			}
```

Comment:
This block will be never executed as in previous if is check `len(status.Applied.Objects.Console.DataGridAuth.Password) > 0`

Reply:
Nope, one check is on spec and another is on status, but I can remove the  len(spec.Objects.Console.DataGridAuth.Password) == 0 
because is checked in the first if